### DISCUSSION TITLE
auto-detection of unavailable master in replicated mode

### DISCUSSION THREAD

**Discussion by jibumathew** [redisson/redisson#5042] [Q&A]
I am getting the error `No master available among the configured addresses, please check your configuration.`  on my master node. My master node is configured as multi-master, which is also configured to replicate from another master node in another data center. This is my configuration:

multi-master yes
active-replica yes
use-fork true
replicaof xx.xx.xx.xx 6379

The `info` command says the current role is ` "active-replica"`. Can someone please help?

I followed this link(https://docs.keydb.dev/docs/multi-master/) to configure my nodes.

**Comment by arulned** [comment]
@jibumathew Did you get this working? I am also using multi-master and need to know if redisson supports it. If not, which library did you use?

**Reply by mrniko** [reply]
@arulned 

It does https://redisson.org/docs/configuration/#multi-cluster-mode

**Reply by arulned** [reply]
@mrniko The multi-cluster-mode requires all individual master nodes. But in my case the single hostname given to me resolves to 3 different IPs like below. 

```
$ ~ → host my.redis-host.com
my.redis-host.com has address x.x.x.105
my.redis-host.com has address x.x.x.99
my.redis-host.com has address x.x.x.106

```
Should I use proxymode?
